datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -452,6 +453,24 @@ private void validateRazEnablement(SdxClusterRequest sdxClusterRequest, Detailed
         }
     }
 
+    private void validateMediumDutyHaEnablement(SdxClusterRequest sdxClusterRequest, DetailedEnvironmentResponse environment) {
+        ValidationResultBuilder validationBuilder = new ValidationResultBuilder();
+        if (sdxClusterRequest.isEnableMediumDutyHA()) {
```

Comment:
Shouldn't this check if the cluster shape requested is medium duty? If it is, then you validate that you have the entitlement.